networkx/algorithms/approximation/tests/test_kcomponents.py

```diff
@@ -222,7 +222,7 @@ def setUp(self):
     def test_size(self):
         for G, A in self.GA:
             n = G.order()
-            s = len(G.edges())+len(A.edges())
+            s = len(list(G.edges())) + len(list(A.edges()))
```

Comment:
Yeah, I think collecting a tools to work with iterators in utils/misc seems like a continuation of what other people have done.

However, would it be a good idea to copy the name `ilen` to be consistent with more_itertools?